node crash on frontend wallet requests

```
panic: runtime error: makeslice: len out of range

goroutine 222704 [running]:
github.com/iotexproject/iotex-core/db.(*countingIndex).Range(0xc00508c5c0, 0x82b, 0xfffffffffffff7f0, 0x14, 0x16113e0, 0xc00508c5c0, 0x0, 0x0)
	/go/apps/iotex-core/db/counting_index.go:163 +0x15c
github.com/iotexproject/iotex-core/blockchain.(*blockDAO).getActionsByAddress(0xc000429560, 0x25f63f4b05f8b001, 0xbb36b01840195de6, 0x178574c7, 0x82b, 0x1e, 0x0, 0x0, 0x1, 0xc000547050, ...)
	/go/apps/iotex-core/blockchain/blockdao.go:411 +0x117
github.com/iotexproject/iotex-core/blockchain.(*blockchain).GetActionsByAddress(0xc0000b0e00, 0xc000cc8600, 0x29, 0x82b, 0x1e, 0x0, 0x0, 0x3, 0x0, 0x4186dc)
	/go/apps/iotex-core/blockchain/blockchain.go:519 +0x176
github.com/iotexproject/iotex-core/api.(*Server).getActionsByAddress(0xc0000b1c00, 0xc000cc8600, 0x29, 0x82b, 0x1e, 0x12bafa0, 0x0, 0x161d260)
	/go/apps/iotex-core/api/api.go:801 +0x127
github.com/iotexproject/iotex-core/api.(*Server).GetActions(0xc0000b1c00, 0x1606840, 0xc000e9dec0, 0xc000e9def0, 0xc0000b1c00, 0x7f2d709086c8, 0xc0002bd8c0)
	/go/apps/iotex-core/api/api.go:191 +0x1f4
github.com/iotexproject/iotex-proto/golang/iotexapi._APIService_GetActions_Handler.func1(0x1606840, 0xc000e9dec0, 0x126a2e0, 0xc000e9def0, 0x1f, 0xc002c929b0, 0xc002c94200, 0xc000127b38)
	/go/pkg/mod/github.com/iotexproject/iotex-proto@v0.2.5/golang/iotexapi/api.pb.go:3304 +0x86
github.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).UnaryServerInterceptor.func1(0x1606840, 0xc000e9dec0, 0x126a2e0, 0xc000e9def0, 0xc003120ee0, 0xc003120f00, 0xc004296ba8, 0x4f73fa, 0x1205d00, 0xc000e9dec0)
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-prometheus@v1.2.0/server_metrics.go:107 +0xad
github.com/iotexproject/iotex-proto/golang/iotexapi._APIService_GetActions_Handler(0x12bafa0, 0xc0000b1c00, 0x1606840, 0xc000e9dec0, 0xc002c92960, 0xc00048a950, 0x1606840, 0xc000e9dec0, 0xc002ac33c0, 0x32)
	/go/pkg/mod/github.com/iotexproject/iotex-proto@v0.2.5/golang/iotexapi/api.pb.go:3306 +0x158
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000a1680, 0x1615400, 0xc0010cc300, 0xc002c94200, 0xc000278930, 0x216db98, 0x0, 0x0, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.21.0/server.go:998 +0x470
google.golang.org/grpc.(*Server).handleStream(0xc0000a1680, 0x1615400, 0xc0010cc300, 0xc002c94200, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.21.0/server.go:1278 +0xda6
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000367f10, 0xc0000a1680, 0x1615400, 0xc0010cc300, 0xc002c94200)
	/go/pkg/mod/google.golang.org/grpc@v1.21.0/server.go:717 +0x9f
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/pkg/mod/google.golang.org/grpc@v1.21.0/server.go:715 +0xa1
```

--- a/api/api.go
+++ b/api/api.go
@@ -723,6 +723,9 @@ func (api *Server) getActionsFromIndex(totalActions, start, count uint64) (*iote
 
 // GetActions returns actions within the range
 func (api *Server) getActions(start uint64, count uint64) (*iotexapi.GetActionsResponse, error) {
+	if count == 0 {
+		return nil, status.Error(codes.InvalidArgument, "count must be greater than zero")
+	}
 	if count > api.cfg.API.RangeQueryLimit {
 		return nil, status.Error(codes.InvalidArgument, "range exceeds the limit")
 	}
@@ -792,6 +795,9 @@ func (api *Server) getSingleAction(actionHash string, checkPending bool) (*iotex
 
 // getActionsByAddress returns all actions associated with an address
 func (api *Server) getActionsByAddress(address string, start uint64, count uint64) (*iotexapi.GetActionsResponse, error) {
+	if count == 0 {
+		return nil, status.Error(codes.InvalidArgument, "count must be greater than zero")
+	}
 	if count > api.cfg.API.RangeQueryLimit {
 		return nil, status.Error(codes.InvalidArgument, "range exceeds the limit")
 	}
@@ -817,6 +823,9 @@ func (api *Server) getActionsByAddress(address string, start uint64, count uint6
 
 // getUnconfirmedActionsByAddress returns all unconfirmed actions in actpool associated with an address
 func (api *Server) getUnconfirmedActionsByAddress(address string, start uint64, count uint64) (*iotexapi.GetActionsResponse, error) {
+	if count == 0 {
+		return nil, status.Error(codes.InvalidArgument, "count must be greater than zero")
+	}
 	if count > api.cfg.API.RangeQueryLimit {
 		return nil, status.Error(codes.InvalidArgument, "range exceeds the limit")
 	}
@@ -845,6 +854,9 @@ func (api *Server) getUnconfirmedActionsByAddress(address string, start uint64, 
 
 // getActionsByBlock returns all actions in a block
 func (api *Server) getActionsByBlock(blkHash string, start uint64, count uint64) (*iotexapi.GetActionsResponse, error) {
+	if count == 0 {
+		return nil, status.Error(codes.InvalidArgument, "count must be greater than zero")
+	}
 	if count > api.cfg.API.RangeQueryLimit {
 		return nil, status.Error(codes.InvalidArgument, "range exceeds the limit")
 	}
@@ -873,6 +885,9 @@ func (api *Server) getActionsByBlock(blkHash string, start uint64, count uint64)
 
 // getBlockMetas returns blockmetas response within the height range
 func (api *Server) getBlockMetas(start uint64, count uint64) (*iotexapi.GetBlockMetasResponse, error) {
+	if count == 0 {
+		return nil, status.Error(codes.InvalidArgument, "count must be greater than zero")
+	}
 	if count > api.cfg.API.RangeQueryLimit {
 		return nil, status.Error(codes.InvalidArgument, "range exceeds the limit")
 	}
@@ -1153,6 +1168,9 @@ func (api *Server) reverseActionsInBlock(blk *block.Block, reverseStart, count u
 }
 
 func (api *Server) getLogsInBlock(filter *LogFilter, start, count uint64) ([]*iotextypes.Log, error) {
+	if count == 0 {
+		return nil, status.Error(codes.InvalidArgument, "count must be greater than zero")
+	}
 	// filter logs within start --> end
 	var logs []*iotextypes.Log
 	end := start + count - 1

--- a/api/api_test.go
+++ b/api/api_test.go
@@ -679,12 +679,12 @@ func TestServer_GetActions(t *testing.T) {
 			},
 		}
 		res, err := svr.GetActions(context.Background(), request)
+		if test.count == 0 {
+			require.Error(err)
+			continue
+		}
 		require.NoError(err)
 		require.Equal(test.numActions, len(res.ActionInfo))
-		if test.numActions == 0 {
-			// returns empty response body in case of no result
-			require.Equal(&iotexapi.GetActionsResponse{}, res)
-		}
 	}
 }
 
@@ -792,15 +792,13 @@ func TestServer_GetUnconfirmedActionsByAddress(t *testing.T) {
 			},
 		}
 		res, err := svr.GetActions(context.Background(), request)
+		if test.count == 0 {
+			require.Error(err)
+			continue
+		}
 		require.NoError(err)
 		require.Equal(test.numActions, len(res.ActionInfo))
-		if test.numActions > 0 {
-			require.Equal(test.address, res.ActionInfo[0].Sender)
-		} else {
-			// returns empty response body in case of no result
-			res.Total = 0
-			require.Equal(&iotexapi.GetActionsResponse{}, res)
-		}
+		require.Equal(test.address, res.ActionInfo[0].Sender)
 	}
 }
 
@@ -825,15 +823,13 @@ func TestServer_GetActionsByBlock(t *testing.T) {
 			},
 		}
 		res, err := svr.GetActions(context.Background(), request)
+		if test.count == 0 {
+			require.Error(err)
+			continue
+		}
 		require.NoError(err)
 		require.Equal(test.numActions, len(res.ActionInfo))
-		if test.numActions > 0 {
-			require.Equal(test.blkHeight, res.ActionInfo[0].BlkHeight)
-		} else {
-			// returns empty response body in case of no result
-			res.Total = 0
-			require.Equal(&iotexapi.GetActionsResponse{}, res)
-		}
+		require.Equal(test.blkHeight, res.ActionInfo[0].BlkHeight)
 	}
 }
 
@@ -854,6 +850,10 @@ func TestServer_GetBlockMetas(t *testing.T) {
 			},
 		}
 		res, err := svr.GetBlockMetas(context.Background(), request)
+		if test.count == 0 {
+			require.Error(err)
+			continue
+		}
 		require.NoError(err)
 		require.Equal(test.numBlks, len(res.BlkMetas))
 		var prevBlkPb *iotextypes.BlockMeta

--- a/blockchain/blockdao/blockdao.go
+++ b/blockchain/blockdao/blockdao.go
@@ -501,6 +501,9 @@ func (dao *blockDAO) getActionsByAddress(addrBytes hash.Hash160, start, count ui
 		return nil, err
 	}
 	total := address.Size()
+	if start >= total {
+		return nil, errors.New("invalid start index")
+	}
 	if start+count > total {
 		count = total - start
 	}

